cachi2/core/package_managers/metayarn.py

```diff
@@ -0,0 +1,25 @@
+from cachi2.core.config import get_config
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import RequestOutput
+from cachi2.core.package_managers import yarn, yarn_classic
+from cachi2.core.package_managers.yarn_classic.main import MissingLockfile, NotV1Lockfile
+from cachi2.core.utils import merge_outputs
+
+
+def fetch_yarn_source(request: Request) -> RequestOutput:
+    """Fetch yarn source."""
+    # Packages could be a mixture of yarn v1 and v2 (at least this is how it
+    # looks now). To preserve this behavior each request is split into individual
+    # packages which are assessed one by one.
+    fetched_packages = []
+    for package in request.packages:
```

Comment:
By implementing a loop to process the input packages here, we are making the same loop that is done in the `fetch_yarn_source` functions for both v1 and v2 redundant.

From one perspective, I'd simplify the `fetch_yarn_source` functions and get rid of the redundancy, since yarn requests will always be initially handled by `metayarn`. On the other hand, keeping them as they are will keep the `yarn` and `yarn_classic` packages independent in case we decide to get rid of `metayarn`, or create another workflow.

Wdyt?